lib/models/environment-variables.coffee

```diff
@@ -214,16 +224,48 @@ getEnvironmentVariablesModel = (deps, opts) ->
 						device: device.id
 					expand: 'device'
 					orderby: 'env_var_name asc'
-		.map (environmentVariable) ->
-
-			# Workaround to the fact that applications environment variables
-			# contains a `name` property, while device environment variables
-			# contains an `env_var_name` property instead.
-			if environmentVariable.env_var_name?
-				environmentVariable.name = environmentVariable.env_var_name
-				delete environmentVariable.env_var_name
-			return environmentVariable
+		.map(fixDeviceEnvVarNameKey)
+		.asCallback(callback)
 
+	###*
+	# @summary Get all device environment variables for an application
+	# @name getAllByApplication
+	# @public
+	# @function
+	# @memberof resin.models.environment-variables.device
+	#
+	# @param {String|Number} nameOrId - application name (string) or id (number)
+	# @fulfil {Object[]} - device environment variables
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.environmentVariables.device.getAllByApplication('MyApp').then(function(environmentVariables) {
+	# 	console.log(environmentVariables);
+	# });
+	#
+	# @example
+	# resin.models.environmentVariables.device.getAllByApplication(999999).then(function(environmentVariables) {
+	# 	console.log(environmentVariables);
+	# });
+	#
+	# @example
+	# resin.models.environmentVariables.device.getAllByApplication('MyApp', function(error, environmentVariables) {
+	# 	if (error) throw error;
+	# 	console.log(environmentVariables)
+	# });
+	###
+	exports.device.getAllByApplication = (nameOrId, callback) ->
+		applicationModel().get(nameOrId).get('id').then (applicationId) ->
+			return pine.get
+				resource: 'device_environment_variable'
+				options:
+					filter:
+						device:
```

Comment:
This way of filtering is deprecated, see https://github.com/resin-io/pinejs-client-js/blob/master/core.coffee#L32